src/db/car/mod.rs

```diff
@@ -14,10 +14,30 @@ use crate::utils::db::car_index::FrameOffset;
 use ahash::HashMap;
 use cid::Cid;
 use lru::LruCache;
-use std::io::{Read, Seek};
+use positioned_io::{ReadAt, Size};
 
-pub trait CarReader: Read + Seek + Send + Sync + 'static {}
-impl<X: Read + Seek + Send + Sync + 'static> CarReader for X {}
+pub trait CarReader: ReadAt + Size + Send + Sync + 'static {}
```

Comment:
`CarReader` is overloaded - can you think of a different name?

Reply:
Yeah, the name is not ideal. Different name? `RandomAccessReader` would do.
Though in the context of car-backed storages it sounds a bit vague.